src/cli/repl/meta_cmd.rs

```diff
@@ -649,8 +669,503 @@ impl<F: LurkField + DeserializeOwned> MetaCmd<F> {
+    /// * If the reduction count is not a number or can't be converted to `u64`
+    fn get_vars_body_and_rc(repl: &Repl<F>, ptcl: Ptr) -> Result<(Vec<Ptr>, Ptr, usize)> {
+        let (io, ..) = repl
+            .eval_expr(ptcl)
+            .with_context(|| "evaluating protocol")?;
+        let ptcl = &io[0];
+
+        let (vars, rest) = repl.store.car_cdr(ptcl)?;
+
+        let Some((vars, None)) = repl.store.fetch_list(&vars) else {
+            bail!("Protocol variables must be a proper list")
+        };
+
+        let (body, rest) = repl.store.car_cdr(&rest)?;
+
+        let (Ptr::Atom(Tag::Expr(ExprTag::Num), idx), _) = repl.store.car_cdr(&rest)? else {
+            bail!("Reduction count must be a Num")
+        };
+        let Some(rc) = repl.store.expect_f(idx).to_u64().map(|u| u as usize) else {
+            bail!("Invalid value for reduction count")
+        };
+        Ok((vars, body, rc))
+    }
+
+    /// Returns a vector containing the elements of a cons-list.
+    ///
+    /// # Errors
+    /// Errors if the the list is not proper or if the number of elements doesn't
+    /// match the expected `len`
+    fn get_args_vec(repl: &Repl<F>, args: &Ptr, len: usize) -> Result<Vec<Ptr>> {
+        let Some((args_vec, None)) = repl.store.fetch_list(args) else {
+            bail!("Protocol arguments must be a proper list")
+        };
+
+        if args_vec.len() != len {
+            bail!("Incompatible number of arguments. Protocol expects {len}")
+        }
+        Ok(args_vec)
+    }
+
+    /// Runs the protocol `body` with an environment built from the bindings
+    /// creted with `vars` and `args` and returns the proof input and the
+    /// post-verify function.
+    ///
+    /// # Errors
+    /// * If the protocol body evaluation fails
+    /// * If the protocol body doesn't evaluate to a pair
+    /// * If the proof input can't be built (first component of the pair is nil)
+    /// * If the proof input is not a proper list with length 6
+    fn get_cek_io_and_post_verify_function(
+        repl: &Repl<F>,
+        vars: Vec<Ptr>,
+        args: Vec<Ptr>,
+        body: Ptr,
+    ) -> Result<(Vec<Ptr>, Ptr)> {
+        let env = vars
+            .into_iter()
+            .zip(args)
+            .fold(repl.store.intern_nil(), |acc, (v, a)| {
+                repl.store.cons(repl.store.cons(v, a), acc)
+            });
+
+        let (io, ..) = repl
+            .eval_expr_with_env(body, env)
+            .with_context(|| "evaluating protocol body")?;
+
+        let Ptr::Tuple2(Tag::Expr(ExprTag::Cons), idx) = &io[0] else {
+            bail!(
+                "Protocol function must return a pair. Got {}",
+                io[0].fmt_to_string(&repl.store, &repl.state.borrow())
+            )
+        };
+        let (pre_verify, post_verify) = repl.store.fetch_2_ptrs(*idx).unwrap();
+
+        if pre_verify.is_nil() {
+            bail!("Pre-verification predicate rejected the input")
```

Comment:
I don't think this should be an error.

Reply:
This should lead to successful rejection of a claim/proof.

Reply:
This is how a proof is successfully rejected. Or do you want `lurk` to exit altogether?

I think an error here is fine btw. It will lead to a unsuccessful execution in the command line.

Reply:
I think there should exist a `verify` predicate that returns `nil` unless the proof is accepted.

Reply:
One would like to be able to reason about claims and their proofs.

Reply:
But meta commands don't return anything. At least not yet.

Reply:
We do need to fix that.